### PR TITLE
Using exception to handle code routine and return value

### PR DESCRIPTION
**Pull request by sharkconi**
1. run_command will raise exception when command execute failed
2. infrasim-init/infrasim-main will handle exception
3. infrasim/socat.py,qemu.py,ipmi.py will handle exception
4. function test will handle exception

Signed-off-by: sharkconi <sharkconi@gmail.com>

### CODE REVIEW

**Comment by sharkconi** [comment]
@xiar @PaynePei @turtle-fly @fub2

**Comment by turtle-fly** [comment]
Great base for InfraSIM error stack. +1

File: infrasim/__init__.py [hunk 1]
```diff
@@ -22,5 +22,29 @@ def run_command(cmd="", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PI
...
     return 0, cmd_result[0]
+
+class InfraSimError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+
+class CommandNotFound(InfraSimError):
+    pass
```
Comment (PaynePei) [inline infrasim/__init__.py]: just placeholder here? will be implemented later?

File: infrasim/__init__.py [hunk 2]
```diff
@@ -22,5 +22,29 @@ def run_command(cmd="", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PI
...
     cmd_return_code = child.returncode
     if cmd_return_code != 0:
-        return -1, cmd_result[1]
+        result = ""
+        if cmd_result[1] is not None:
+            result = cmd + ":" + cmd_result[1]
+        else:
+            result = cmd
+        logger.error(result)
+        raise CommandRunFailed(result)
```
Comment (PaynePei) [inline infrasim/__init__.py]: zan!

File: infrasim/__init__.py (hunk 1 above)
Comment (sharkconi) [inline infrasim/__init__.py]: Yes, this can be implement or not, depends on how much we need to handle the error

File: infrasim/__init__.py (hunk 2 above)
Comment (sharkconi) [inline infrasim/__init__.py]: Yes.  This place can grab the full error message and command execute return code

**Pull request merged**